R/CreatePackageVignette.R

```diff
@@ -0,0 +1,101 @@
+#' @title pkgnet Report as Vignette
+#' @name CreatePackageVignette
+#' @description Create pkgnet package report as an Rmarkdown vignette. This
+#'    vignette is able to be rendered with the
+#'    \code{\link[knitr:vignette_engines]{knitr::rmarkdown}} vignette engine
+#'    into HTML vignettes upon package building. It is also compatible with
+#'    \code{\link[pkgdown:build_articles]{pkgdown}} sites.
+#' @param pkg_name (string) name of a package
+#' @param pkg_path (string) The path to the package repository. If given, coverage
+#'                 will be calculated for each function. \code{pkg_path} can be an
+#'                 absolute or relative path.
+#' @param pkg_reporters (list) a list of initialized package reporters
+#' @param vignette_path (string) The path and filename of the output vignette
+#'    file. The default assumes your working directory is the package root.
+#' @importFrom rlang enexpr
+#' @importFrom assertthat assert_that is.string is.readable
+#' @importFrom tools file_ext
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
+                                  , pkg_path = NULL
+                                  , pkg_reporters = list(
+                                      DependencyReporter$new()
+                                      , FunctionReporter$new()
+                                  )
+                                  , vignette_path = file.path("vignettes"
+                                                              , "pkgnet_report.Rmd")
+                                 ) {
+
+    # Capture pkg_reporters expression for later injection into Rmd
+    pkg_reporters_expr <- rlang::enexpr(pkg_reporters)
```

Comment:
In the spirt of this package, I must ask if this function is worth the new package dependency, especially since `rlang` warns that their "API is still maturing."

https://github.com/r-lib/rlang

Reply:
Great question. I did look on [this webpage on rlang's website](https://rlang.r-lib.org/reference/lifecycle.html) which says that `enexpr` is a stable function that should not change.

Reply:
I want to unresolve this conversation. I have the same question as Brian (but from a different angle). What does this function do? Do we really really really need it? How is it different from `as.expression()`?

`rlang` may look harmless from its `Imports`, but it's C code is kind of hard to build. I've had a lot of trouble compiling `rlang` in the past.

Reply:
``` r

library(pkgnet)

func1 <- function(x) {
    y <- as.expression(x)
    return(deparse(y))
}

func1(list(DependencyReporter$new()))
#> [1] "expression(<environment>)"

func2 <- function(x) {
    y <- rlang::enexpr(x)
    return(deparse(y))
}

func2(list(DependencyReporter$new()))
#> [1] "list(DependencyReporter$new())"
```

Reply:
`enexpr` is specifically for capturing inputs to functions as expressions without having evaluated them.

Reply:
also hey guess what

<img width="400" alt="Screen Shot 2019-03-14 at 8 39 41 AM" src="https://user-images.githubusercontent.com/2721979/54361322-cb072e80-4634-11e9-8c95-5b73b05242d6.png">

Reply:
@bburns632 this totally could be an example for our talk ^

Reply:
WOW!!!!!!!

ok pretty compelling argument